packages/nx-plugin/src/generators/generator/files/generator/__fileName__/generator.ts__tmpl__

```diff
@@ -23,7 +23,10 @@ function normalizeOptions(tree: Tree, options: <%= className %>GeneratorSchema):
     ? `${names(options.directory).fileName}/${name}`
     : name;
   const projectName = projectDirectory.replace(new RegExp('/', 'g'), '-');
-  const projectRoot = `${getWorkspaceLayout(tree).libsDir}/${projectDirectory}`;
+  const projectRoot =
+    getWorkspaceLayout(tree).libsDir === '.'
+      ? '.'
+      : `${getWorkspaceLayout(tree).libsDir}/${projectDirectory}`;
```

Comment:
What necessitated this change? For a workspace with `libsDir` set to '.', running a generator should create a new project that's not in the root. It would be in `./${projectDirectory}`, which seems equivalent?

Reply:
actually, projectDirectory contains the library name. if not rootProject, it would be libs/custom-directory/library-name. without the condition, if rootProject, it would be `library-name`.